test/Table_Tests/src/Common_Table_Operations/Join/Union_Spec.enso

```diff
@@ -245,29 +290,79 @@ run_union_tests group_builder setup call_union =
 
         t3 = call_union [t1, t2]
         expect_column_names ["A"] t3
+        Problems.assume_no_problems t3
         t3.at "A" . value_type . is_floating_point . should_be_true
         t3.at "A" . to_vector . should_equal [0, 1, 2, 1.0, 2.0, 2.5]
 
-        # Specific type tests that apply to in-memory. Database behaviour is up to implementation.
-        if setup.is_database.not then
-            t4 = table_builder [["A", [2^100, 2^10, 2]]]
-            t4.at "A" . value_type . should_be_a (Value_Type.Decimal ...)
+    group_builder.specify "should find a common type that will fit the merged columns (numeric + Boolean)" <|
+        t1 = table_builder [["A", [0, 1, 20]]]
+        t2 = table_builder [["A", [True, False, True]]]
+
+        t1.at "A" . value_type . is_integer . should_be_true
+        t2.at "A" . value_type . should_equal Value_Type.Boolean
+
+        t3 = call_union [t1, t2]
+        expect_column_names ["A"] t3
+        Problems.assume_no_problems t3
+        t3.at "A" . value_type . is_integer . should_be_true
+        t3.at "A" . to_vector . should_equal [0, 1, 20, 1, 0, 1]
+
+        t4 = table_builder [["A", [1.5, 0.0, 2.0]]]
+        t5 = call_union [t2, t4]
+        Problems.assume_no_problems t5
+        t5.at "A" . value_type . is_floating_point . should_be_true
+        t5.at "A" . to_vector . should_equal [1.0, 0.0, 1.0, 1.5, 0.0, 2.0]
+
+    ## TODO once the https://github.com/enso-org/enso/issues/8355 is implemented, the behaviour of this test will probably change:
+       When merging Float and big Integers, maybe we shall convert to BigDecimal, not to Float?
```

Comment:
We should discuss this. Conversion to Decimal would be the most general way to do it, but Decimals are a lot slower and we might want to prevent conversion to it in cases like this.

Reply:
Fair point. I guess we can decide to not do it in that case.